engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -58,6 +58,29 @@ class PeriodicProcessService(
 
   private implicit val localDateOrdering: Ordering[LocalDateTime] = Ordering.by(identity[ChronoLocalDateTime[_]])
 
+  def getScenarioActivitiesSpecificToPeriodicProcess(
+      processIdWithName: ProcessIdWithName
+  ): Future[List[ScenarioActivity]] =
+    scheduledProcessesRepository
+      .getSchedulesState(processIdWithName.name)
+      .run
+      .map(_.groupedByPeriodicProcess)
+      .map(_.flatMap(_.deployments))
+      .map(_.map { deployment =>
+        ScenarioActivity.PerformedScheduledExecution(
+          scenarioId = ScenarioId(processIdWithName.id.value),
+          scenarioActivityId = ScenarioActivityId.random,
+          user = ScenarioUser(None, UserName("Nussknacker"), None, None),
```

Comment:
shouldn't we have defined `UserName.internalNuUser` (or special user, or sth like that)? And use it in all places we need sth like this?

Reply:
Done, there is not ScenarioUser.internalNuUser. Is is the only usage ATM, in all other places we have actual, logged-in user (Here it would be possible too - we could use the user that deployed the scenario instead of "Nussknacker" - but we don't have that information available ATM)